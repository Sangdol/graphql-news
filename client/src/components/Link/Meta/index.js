import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Wrapper = styled.div`
  display: flex;
  margin-top: 0.25em;
  color: #848584;

  a {
    color: inherit;

    &:hover {
      color: #ff6600;
    }
  }
`;

const Meta = ({ author, commentsLength }) => (
  <Wrapper>
    <span>
      by <a href="#">{author}</a> 3 hours ago
    </span>
    <span style={{ margin: '0 0.25em' }}>|</span>
    <span>
      <a href="#">hide</a>
    </span>
    <span style={{ margin: '0 0.25em' }}>|</span>
    <span>
      <a href="#">{commentsLength} comments</a>
    </span>
  </Wrapper>
);

Meta.propTypes = {
  author: PropTypes.string.isRequired,
  commentsLength: PropTypes.number.isRequired,
};

export default Meta;
