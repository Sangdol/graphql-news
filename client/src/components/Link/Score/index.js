import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import UpvoteButton from './upvoteButton';
import DownvoteButton from './downvoteButton';
import ScoreCount from './scoreCount';

const Wrapper = styled.div`
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  margin: 0 1em 0 0;
`;

const Score = ({ _id, score }) => (
  <Wrapper>
    <UpvoteButton _id={_id} />
    <ScoreCount _id={_id} score={score} />
    <DownvoteButton _id={_id} />
  </Wrapper>
);

Score.propTypes = {
  _id: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
  client: PropTypes.object.isRequired,
};

export default Score;
