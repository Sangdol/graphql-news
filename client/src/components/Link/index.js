import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.a`
  color: #34495e;
  text-decoration: none;
  font-size: 1.2em;
`;

const Url = styled.a`
  font-size: 0.8em;
  color: #848584;
  margin: 0 0 0 0.25em;
  text-decoration: none;
`;

const Link = ({ author, url, description, commentsLength, score }) => (
  <div style={{ display: 'flex', alignItems: 'center' }}>
    <Score score={score} />
    <div style={{ display: 'inline-flex', flexDirection: 'column' }}>
      <div>
        <Title href="#">{description}</Title>
        <Url href="#">({url.replace(/(^\w+:|^)\/\//, '')})</Url>
      </div>
      <Meta author={author} commentsLength={commentsLength} />
    </div>
  </div>
);

Link.propTypes = {
  author: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  commentsLength: PropTypes.number.isRequired,
  score: PropTypes.number.isRequired,
};

const ScoreContainer = styled.div`
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  margin: 0 1em 0 0;
`;

const ScoreTrigger = styled.div`
  font-size: 0.6em;
  cursor: pointer;
`;

const Arrow = styled.span`
  display: inline-block;
  color: #848584;
`;

const ScoreCount = styled.span`
  font-weight: bold;
  color: #ff6600;
  margin: 0.2em 0;
`;

const Score = ({ score }) => (
  <ScoreContainer>
    <ScoreTrigger>
      <Arrow style={{ color: '#FF6600' }}>▲</Arrow>
    </ScoreTrigger>

    <ScoreCount>{score}</ScoreCount>

    <ScoreTrigger>
      <Arrow>▼</Arrow>
    </ScoreTrigger>
  </ScoreContainer>
);

Score.propTypes = {
  score: PropTypes.number.isRequired,
};

const MetaContainer = styled.div`
  display: flex;
  margin-top: 0.25em;
  color: #848584;

  a {
    &:hover {
      color: #ff6600;
    }
  }
`;

const Separator = styled.span`
  margin: 0 0.25em;
`;

const Anchor = styled.a`
  color: inherit;
`;

const Meta = ({ author, commentsLength }) => (
  <MetaContainer>
    <span>
      by <Anchor href="#">{author}</Anchor> 3 hours ago
    </span>
    <Separator>|</Separator>
    <span>
      <Anchor href="#">hide</Anchor>
    </span>
    <Separator>|</Separator>
    <span>
      <Anchor href="#">{commentsLength} comments</Anchor>
    </span>
  </MetaContainer>
);

Meta.propTypes = {
  author: PropTypes.string.isRequired,
  commentsLength: PropTypes.number.isRequired,
};

export default Link;
