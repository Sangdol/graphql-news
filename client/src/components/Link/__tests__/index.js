import React from 'react';
import TestRenderer from 'react-test-renderer';
import Link from '../index';

describe('the <Link /> component', async () => {
  test('should look like this', async () => {
    const tree = TestRenderer.create(
      <Link
        author="anonymous"
        url="https://google.com"
        description="The internet's front door"
        comments={70}
        score={275}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
