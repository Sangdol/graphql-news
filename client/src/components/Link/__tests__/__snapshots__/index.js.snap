// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <Link /> component should look like this 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <div
    className="sc-htpNat iNpPag"
  >
    <div
      className="sc-bxivhb iLWNZv"
    >
      <span
        className="sc-ifAKCX coRGWX"
        style={
          Object {
            "color": "#FF6600",
          }
        }
      >
        ▲
      </span>
    </div>
    <span
      className="sc-EHOje juxLGl"
    >
      275
    </span>
    <div
      className="sc-bxivhb iLWNZv"
    >
      <span
        className="sc-ifAKCX coRGWX"
      >
        ▼
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "display": "inline-flex",
        "flexDirection": "column",
      }
    }
  >
    <div>
      <a
        className="sc-bdVaJa kPQtlt"
        href="#"
      >
        The internet's front door
      </a>
      <a
        className="sc-bwzfXH fNPCXg"
        href="#"
      >
        (
        google.com
        )
      </a>
    </div>
    <div
      className="sc-bZQynM dyAiHf"
    >
      <span>
        by 
        <a
          className="sc-htoDjs eEVTBG"
          href="#"
        >
          anonymous
        </a>
         3 hours ago
      </span>
      <span
        className="sc-gzVnrw bAZDPE"
      >
        |
      </span>
      <span>
        <a
          className="sc-htoDjs eEVTBG"
          href="#"
        >
          hide
        </a>
      </span>
      <span
        className="sc-gzVnrw bAZDPE"
      >
        |
      </span>
      <span>
        <a
          className="sc-htoDjs eEVTBG"
          href="#"
        >
          70
           comments
        </a>
      </span>
    </div>
  </div>
</div>
`;
